build: warnings from CUDA

<!--
If you have a question rather than reporting a bug please go to https://forum.opencv.org where you get much faster responses.
If you need further assistance please read [How To Contribute](https://github.com/opencv/opencv/wiki/How_to_contribute).

This is a template helping you to create an issue which can be processed as quickly as possible. This is the bug reporting section for the OpenCV library.
-->

##### System information (version)
<!-- Example
- OpenCV => 4.2
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2017
-->


- OpenCV => recent next branch  ( 593a376 ) + recent contrib next branch ( d980cc3 )
- Operating System / Platform => Jetson TX2/TX1/Xavier/Nano
- Compiler => GCC 7.5.0/5.4.0

##### Detailed description

After #3370 , I see following warnings from `cudalegacy` module
I'll send a patch shortly

```
[ 68%] Building CXX object modules/cudalegacy/CMakeFiles/opencv_cudalegacy.dir/src/fgd.cpp.o
/opencv_contrib/modules/cudalegacy/src/fgd.cpp: In function ‘int {anonymous}::findForegroundRegions(cv::cuda::GpuMat&, cv::Mat&, std::vector<std::vector<cv::Point_<int> > >&, CvMemStorage*, const cv::cuda::FGDParams&)’:
/opencv_contrib/modules/cudalegacy/src/fgd.cpp:383:27: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
         for (int i = 0; i < contours.size(); ++i)
                           ^
/opencv_contrib/modules/cudalegacy/src/fgd.cpp:388:57: warning: suggest parentheses around ‘&&’ within ‘||’ [-Wparentheses]
             if (brect.area() < params.minArea || isHole && params.is_obj_without_holes)
                                                         ^
/opencv_contrib/modules/cudalegacy/src/fgd.cpp: At global scope:
/opencv_contrib/modules/cudalegacy/src/fgd.cpp:370:45: warning: unused parameter ‘storage’ [-Wunused-parameter]
                               CvMemStorage* storage, const FGDParams& params)
                                             ^
/opencv_contrib/modules/cudalegacy/src/fgd.cpp:348:10: warning: ‘void {anonymous}::seqToContours(CvSeq*, CvMemStorage*, cv::OutputArrayOfArrays)’ defined but not used [-Wunused-function]
     void seqToContours(CvSeq* _ccontours, CvMemStorage* storage, OutputArrayOfArrays _contours)
          ^

```
##### Steps to reproduce

```
cmake -DOPENCV_EXTRA_MODULES_PATH=/opencv_contrib/modules -DBUILD_EXAMPLES=true -DOPENCV_ENABLE_NONFREE=true -DWITH_CUDA=ON -DPYTHON2_EXECUTABLE= -DCMAKE_BUILD_TYPE=Release -DCPU_DISPATCH=NEON_DOTPROD ..
make all
```

##### Issue submission checklist

 - [x] I report the issue, it's not a question
   <!--
   OpenCV team works with forum.opencv.org, Stack Overflow and other communities
   to discuss problems. Tickets with questions without a real issue statement will be
   closed.
   -->
 - [x] I checked the problem with documentation, FAQ, open issues,
       forum.opencv.org, Stack Overflow, etc and have not found any solution
   <!--
   Places to check:
   * OpenCV documentation: https://docs.opencv.org
   * FAQ page: https://github.com/opencv/opencv/wiki/FAQ
   * OpenCV forum: https://forum.opencv.org
   * OpenCV issue tracker: https://github.com/opencv/opencv/issues?q=is%3Aissue
   * Stack Overflow branch: https://stackoverflow.com/questions/tagged/opencv
   -->
 - [x] I updated to the latest OpenCV version and the issue is still there
   <!--
   master branch for OpenCV 4.x and 3.4 branch for OpenCV 3.x releases.
   OpenCV team supports only the latest release for each branch.
   The ticket is closed if the problem is not reproduced with the modern version.
   -->
 - [x] There is reproducer code and related data files: videos, images, onnx, etc
   <!--
   The best reproducer -- test case for OpenCV that we can add to the library.
   Recommendations for media files and binary files:
   * Try to reproduce the issue with images and videos in opencv_extra repository
     to reduce attachment size
   * Use PNG for images, if you report some CV related bug, but not image reader
     issue
   * Attach the image as an archive to the ticket, if you report some reader issue.
     Image hosting services compress images and it breaks the repro code.
   * Provide ONNX file for some public model or ONNX file with random weights,
     if you report ONNX parsing or handling issue. Architecture details diagram
     from netron tool can be very useful too. See https://lutzroeder.github.io/netron/
   -->

--- a/modules/cudalegacy/src/fgd.cpp
+++ b/modules/cudalegacy/src/fgd.cpp
@@ -345,29 +345,8 @@ namespace
 
 namespace
 {
-    void seqToContours(CvSeq* _ccontours, CvMemStorage* storage, OutputArrayOfArrays _contours)
-    {
-        Seq<CvSeq*> all_contours(cvTreeToNodeSeq(_ccontours, sizeof(CvSeq), storage));
-
-        size_t total = all_contours.size();
-
-        _contours.create((int) total, 1, 0, -1, true);
-
-        SeqIterator<CvSeq*> it = all_contours.begin();
-        for (size_t i = 0; i < total; ++i, ++it)
-        {
-            CvSeq* c = *it;
-            ((CvContour*)c)->color = (int)i;
-            _contours.create((int)c->total, 1, CV_32SC2, (int)i, true);
-            Mat ci = _contours.getMat((int)i);
-            CV_Assert( ci.isContinuous() );
-            cvCvtSeqToArray(c, ci.data);
-        }
-    }
-
-
     int findForegroundRegions(GpuMat& d_foreground, Mat& h_foreground, std::vector< std::vector<Point> >& foreground_regions,
-                              CvMemStorage* storage, const FGDParams& params)
+                              const FGDParams& params)
     {
         int region_count = 0;
 
@@ -380,12 +359,12 @@ namespace
         CV_Assert(contours.size() > 0);
 
         // adding top-level contours to results, filtering by size
-        for (int i = 0; i < contours.size(); ++i)
+        for (size_t i = 0; i < contours.size(); ++i)
         {
             const std::vector<cv::Point> & cnt = contours[i];
             const cv::Rect brect = cv::boundingRect(cnt);
             bool isHole = hierarchy[i][3] >= 0; // contour with parent is hole
-            if (brect.area() < params.minArea || isHole && params.is_obj_without_holes)
+            if (brect.area() < params.minArea || (isHole && params.is_obj_without_holes))
                 continue;
             foreground_regions.push_back(cnt);
         }
@@ -593,19 +572,14 @@ namespace
         Ptr<cuda::Filter> dilateFilter_;
         Ptr<cuda::Filter> erodeFilter_;
 #endif
-
-        CvMemStorage* storage_;
     };
 
     FGDImpl::FGDImpl(const FGDParams& params) : params_(params), frameSize_(0, 0)
     {
-        storage_ = cvCreateMemStorage();
-        CV_Assert( storage_ != 0 );
     }
 
     FGDImpl::~FGDImpl()
     {
-        cvReleaseMemStorage(&storage_);
     }
 
     void FGDImpl::apply(InputArray _frame, OutputArray fgmask, double)
@@ -621,7 +595,6 @@ namespace
         CV_Assert( curFrame.type() == CV_8UC3 || curFrame.type() == CV_8UC4 );
         CV_Assert( curFrame.size() == prevFrame_.size() );
 
-        cvClearMemStorage(storage_);
         foreground_regions_.clear();
         foreground_.setTo(Scalar::all(0));
 
@@ -636,7 +609,7 @@ namespace
 #endif
 
         if (params_.minArea > 0 || params_.is_obj_without_holes)
-            findForegroundRegions(foreground_, h_foreground_, foreground_regions_, storage_, params_);
+            findForegroundRegions(foreground_, h_foreground_, foreground_regions_, params_);
 
         // Check ALL BG update condition:
         const double BGFG_FGD_BG_UPDATE_TRESH = 0.5;

